src/pages/academy/grading/Grading.tsx

```diff
@@ -48,22 +53,52 @@ const Grading: React.FC = () => {
 
   const [pageSize, setPageSize] = useState(10);
   const [showAllSubmissions, setShowAllSubmissions] = useState(false);
+  const [refreshQueried, setRefreshQueried] = useState(false); // for callback (immediately becomes false)
+  const [animateRefresh, setAnimateRefresh] = useState(false); // for animation (becomes false on animation end)
+  const [submissions, setSubmissions] = useState<GradingOverview[]>([]);
 
   const dispatch = useDispatch();
+  const allColsSortStates = useTypedSelector(state => state.workspaces.grading.allColsSortStates);
+  const hasLoadedBefore = useTypedSelector(state => state.workspaces.grading.hasLoadedBefore);
+
   const updateGradingOverviewsCallback = useCallback(
     (page: number, filterParams: Object) => {
+      // Prevents es-lint missing dependency warning
+      if (refreshQueried) {
+        return setRefreshQueried(false);
+      }
+
+      dispatch(setGradingHasLoadedBefore());
+      dispatch(increaseRequestCounter());
```

Comment:
Can I check what's the request counter for?

Reply:
Loading Overlay. Also, I've added the relevant changes to the other comments that I didn't comment on in the latest commit 😄